proxycfg: move each handler into a seprate file

dnephin opened the pull request:
Branched from #9489

There is no interaction between these handlers, so splitting them into separate files makes it easier to discover the full implementation of each kindHandler.

This PR does not change any functionality. It only moves the handlers into their own files.

dnephin commented on agent/proxycfg/connect_proxy.go:
```diff
@@ -0,0 +1,355 @@
+	return snap, nil
+}
+
+// reducedUpstreamConfig represents the basic opaque config values that are now
```
TODO: Move to upstreams file.

freddygv reviewed (approved):
LGTM. Left one question above the move to an upstreams specific file, but that land in a follow-up if that's what you prefer.

dnephin commented:
I re-applied all the changes again (to make sure I did not miss anything in the diff). I also noticed that an earlier PR accidentally re-introduced a method that had been deleted, so I fixed that and deleted the method.

hc-github-team-consul-core commented:
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/391415.
